src/protocol/mod.rs

```diff
@@ -59,6 +66,25 @@ where
             params: None,
         }
     }
+    /// set which serialization method is used in the body
+    /// if feature "extra_formats" is not enabled, this function
+    /// does not do anything and json is used.
+    #[cfg(any(test, feature = "extra_formats"))]
+    pub fn serializer(mut self, format: MessageFormat) -> Self {
+        use MessageFormat::*;
+        let format_name = match format {
+            Json => "application/json",
+            Yaml => "application/x-yaml",
+            Pickle => "application/x-python-serialize",
+            MsgPack => "application/x-msgpack",
+        };
+        self.message.properties.content_type = format_name.into();
+        self
+    }
+    #[cfg(not(any(test, feature = "extra_formats")))]
+    pub fn serializer(self, _format: MessageFormat) -> Self {
+        self
+    }
```

Comment:
Nice!

Do you think it would be better to just omit this method when the `extra_formats` feature is not enabled?